ingest-app/src/cmr/ingest/api/subscriptions.clj

```diff
@@ -18,30 +19,60 @@
+                                                   request-context
+                                                   headers)
+        ;; Log the ingest attempt before the save
         _ (info (format "Ingesting subscription %s from client %s"
                         (api-core/concept->loggable-string concept-with-user-id)
                         (:client-id request-context)))
-        save-subscription-result (ingest/save-subscription request-context concept-with-user-id)]
+        save-subscription-result (ingest/save-subscription request-context
+                                                           concept-with-user-id)]
     ;; Log the successful ingest, with the metadata size in bytes.
     (api-core/log-concept-with-metadata-size concept-with-user-id request-context)
     (api-core/generate-ingest-response headers save-subscription-result)))
 
+(defn ingest-subscription
+  "Processes a request to create or update a subscription. Note, this will allow
+  unlimited subscriptions to be created by users for themselfs. Revisit if this
+  Becomes a problem"
+  [provider-id native-id request]
+  (let [{:keys [body content-type headers request-context]} request]
+    (common-enabled/validate-write-enabled request-context "ingest")
+    (lt-validation/validate-launchpad-token request-context)
+    (api-core/verify-provider-exists request-context provider-id)
+    (let [concept (api-core/body->concept! :subscription provider-id native-id
+                                           body
+                                           content-type
+                                           headers)
+          subscription-user (:SubscriberId (json/decode (:medata concept) true))
+          token-user (api-core/get-user-id request-context headers)]
+      (if (= token-user subscription-user)
+        (warn (format (str "ACLs were bypassed because the token account %s "
+                      "matched the subscription user %s in the metadata.")
+                      token-user
+                      subscription-user))
+        (do
+          (info (format (str "ACLs were checked because the token user %s is "
+                        "not the same as the subscription user %s in the "
+                        "metadata.")
+                        token-user
+                        subscription-user))
+          (acl/verify-ingest-management-permission request-context
+                                                   :update
+                                                   :provider-object
+                                                   provider-id)
+          (acl/verify-subscription-management-permission request-context
+                                                         :update
+                                                         :provider-object
+                                                         provider-id)))
+      (perform-subscription-ingest concept request-context headers))))
```

Comment:
Nested `let`s are a code smell and this could easily be broken off into another function. I'm really not okay with this abuse of the `let` block and am happy to pair with you if you'd like